lib/DBDish/Connection.pm6

```diff
@@ -49,6 +52,22 @@ method new(*%args) {
 
 method prepare(Str $statement, *%args) { ... }
 
+# Many DBs can only use the connection by a single thread at a time.
+# Lock this with a fast CAS operation for minimal overhead. Don't try to
+# wait for it to be free, just notify the user about the problem.
+method is-connection-available() {$!connection-lock == 0}
```

Comment:
I'm not sure why this method is needed. Is this just for testing? If so, I think the behavioral test covers it well enough.

Reply:
Indeed. It's not needed.

Removed.